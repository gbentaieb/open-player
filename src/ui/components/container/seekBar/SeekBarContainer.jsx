import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { seekTo } from '../../../actions/CoreActions';
import SeekBar from '../../presentational/seekBar/SeekBar';

export function mapStateToProps(state) {
  return {
    currentTime: state.core.currentTime,
    startTime: state.core.startTime,
    endTime: state.core.endTime,
  };
}

class SeekBarContainer extends Component {
  static propTypes = {
    currentTime: PropTypes.number,
    startTime: PropTypes.number,
    endTime: PropTypes.number,
    seekTo: PropTypes.func.isRequired,
  }

  static defaultProps = {
    currentTime: 0,
    startTime: 0,
    endTime: Infinity,
  }

  render() {
    return (
      <SeekBar
        currentTime={this.props.currentTime}
        startTime={this.props.startTime}
        endTime={this.props.endTime}
        seekTo={this.props.seekTo}
      />
    );
  }
}

export default connect(
  mapStateToProps,
  { seekTo },
)(SeekBarContainer);
