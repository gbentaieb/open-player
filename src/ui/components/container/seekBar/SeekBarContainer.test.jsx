import React from 'react';
import { createMockStore } from 'redux-test-utils';
import SeekBarContainer, { mapStateToProps } from './SeekBarContainer';

const defaultStore = {
  core: {
    currentTime: 2,
    startTime: 1,
    endTime: 3,
  },
};

describe('SeekBarContainer', () => {
  test('SeekBarContainer > Snapshot', () => {
    const store = createMockStore(defaultStore);
    const wrapper = shallow(<SeekBarContainer store={store} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('SeekBarContainer > mapStateToProps', () => {
    const state = defaultStore;

    expect(mapStateToProps(state)).toEqual(state.core);
  });
});
