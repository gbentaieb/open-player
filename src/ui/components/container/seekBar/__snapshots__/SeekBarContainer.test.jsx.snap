// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeekBarContainer SeekBarContainer > Snapshot 1`] = `
<SeekBarContainer
  currentTime={2}
  endTime={3}
  seekTo={[Function]}
  startTime={1}
  store={
    Object {
      "dispatch": [Function],
      "getAction": [Function],
      "getActions": [Function],
      "getState": [Function],
      "isActionDispatched": [Function],
      "isActionTypeDispatched": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getAction": [Function],
        "getActions": [Function],
        "getState": [Function],
        "isActionDispatched": [Function],
        "isActionTypeDispatched": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": undefined,
    }
  }
/>
`;
